pkg/controller/jobframework/setup.go

```diff
@@ -110,3 +157,88 @@ func SetupIndexes(ctx context.Context, indexer client.FieldIndexer, opts ...Opti
 		return nil
 	})
 }
+
+func isAPIAvailable(mgr ctrl.Manager, rcApiNames []string, toApiNames []string) bool {
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+	apiGroupResources, err := restmapper.GetAPIGroupResources(discoveryClient)
+	exitOnError(err, "unable to get API group resources")
+
+	restMapper := restmapper.NewDiscoveryRESTMapper(apiGroupResources)
+	checkAPIs := func(apiNames []string) bool {
+		for _, apiName := range apiNames {
+			_, err = restMapper.KindFor(schema.GroupVersionResource{Resource: apiName})
```

Comment:
Hmm, given we are using the GroupVersion directly, can we use https://github.com/ray-project/kuberay/blob/5bf70e8e4e0421c0b60a82cef43941a1e66aa714/ray-operator/apis/ray/v1/groupversion_info.go#L13 directly?